CI: blocking lint and more lint targets

jbertran opened the pull request:
#### WHAT

Make lint blocking and fix python linting issues.

Add linting targets:
* YAML
* Shell

#### WHY

Blocking lint tests is good. So is more linters.

Commit:
ci: fix python linting issues

Issue GH-637

bert-e commented:
# Hello jbertran,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @cmonfort

* @alxf

* @jbertran

* @ballot-scality

* @carlito-scality

* @nicolast

* @tcarmet

* @zempashi

* @sayf-eddine-scality

* @gdemonet

* @monpote

* @alexandre-allard-scality

* @lucieleonard

* @slaperche-scality

* @thomasdanan

* @ramanduh

* @nootal

* @teddyandrieux

slaperche-scality commented on tox.ini:
```diff
@@ -16,6 +16,15 @@ deps =
 
+[testenv:lint-yaml]
+whitelist_externals = bash
+basepython = python3.6
```
Do you need to specify `basepython`? It's not inherited from `testenv`?

jbertran replied:
Fixed

charlesprost commented on salt/_states/containerd.py:
```diff
@@ -114,5 +90,45 @@ def image_managed(name, archive_path=None):
             ret['comment'] = 'Failed to pull image'
+        return ret
+
+    return ret
```
Maybe `return ret` only once?

jbertran replied:
I really dislike the control flow of these dict-filling steps. Yes, you're absolutely right.

charlesprost commented on salt/_states/containerd.py:
```diff
@@ -114,5 +90,45 @@ def image_managed(name, archive_path=None):
+
 
+def manage_archive_path(name, archive_path, ret):
+    '''Pull image from archive at the specified path
```
Missing endpoint?

jbertran replied:
Done

charlesprost commented on salt/_states/containerd.py:
```diff
@@ -114,5 +90,45 @@ def image_managed(name, archive_path=None):
+    name
+        Tag or digest of the image
+    archive_path
+        path to the local Docker archive of the image to load if
```
capitalization of `path`?

jbertran replied:
Done

charlesprost commented on salt/_states/containerd.py:
```diff
@@ -114,5 +90,45 @@ def image_managed(name, archive_path=None):
+    else:
+        ret['comment'] = 'Failed to import archive'
+
+    return ret
```
Same for `return ret`?

charlesprost reviewed (dismissed):
minor issues to be fixed

NicolasT commented on make/lint.mk:
```diff
@@ -5,5 +5,7 @@
+	tox -e lint-yaml
+	tox -e lint-shell
 
 .PHONY: lint-local
```
There's no `lint-local` target?

jbertran replied:
Indeed :o

NicolasT commented on salt/_modules/containerd.py:
```diff
@@ -7,7 +7,7 @@
 import salt.utils.path
 
-log = logging.getLogger(__name__)
+LOG = logging.getLogger(__name__)
```
Most other (upstream) Salt modules use `log` i.s.o. `LOG`. Maybe this should be an exception in the linter.

slaperche-scality replied:
On the other hand, it's pretty common to have UPPERCASE for global variable.

So depends if we want clean code or Salt-like code here.

But if we add an exception to the linter, it should be only for Salt code, as we probably don't want these kind of weird decision to be applied to Python script in general

jbertran replied:
Leaving it as `log` for now since I'm dropping Python linting

slaperche-scality replied:
Why reverting that change? You'll have to reintroduce it when we re-enable linting anyway, so you could keep it.

jbertran replied:
It's trivial enough to reproduce, and if I want to see if I can work in the Salt-specific exception.

NicolasT commented on salt/_states/containerd.py:
```diff
@@ -114,5 +90,45 @@ def image_managed(name, archive_path=None):
+    return ret
+
 
+def manage_archive_path(name, archive_path, ret):
```
This naming seems not in line with other states. `manage` is active (i.e. a module).

If the intent is for this to be not 'public API of the state module' make it `_manage_archive_path` or something, though even then, even utility procedures in a state should be... a state-like thing.

jbertran replied:
Reverting this since I'm dropping Python linting.

NicolasT commented on salt/_states/containerd.py:
```diff
@@ -114,5 +90,45 @@ def image_managed(name, archive_path=None):
+    return ret
+
 
+def manage_archive_path(name, archive_path, ret):
```
What's this `ret` thing?

jbertran replied:
Reverted. Getting lost in salt return dict side effects.

NicolasT commented on tox.ini:
```diff
@@ -7,14 +7,30 @@ skip_install = true
+basepython = python3.6
+skip_install = true
+deps =
+    -r{toxinidir}/lint-requirements.txt
```
Now you're installing all lint deps in 3 different venvs. Either

- List linter-specific deps here (I think that'd be my preference)
- Set up Tox to re-use the same env between the 3 jobs

jbertran replied:
lint deps are minimal, so linter-specific it is.

bert-e commented:
# Conflict




There is a conflict between your branch `improvement/GH-637-blocking-lint-and-more-lint-targets` and the
destination branch `development/2.0`.

Please resolve the conflict on **the feature branch** (`improvement/GH-637-blocking-lint-and-more-lint-targets`).

```bash
 $ git fetch
 $ git checkout origin/improvement/GH-637-blocking-lint-and-more-lint-targets
 $ git merge origin/development/2.0
 $ # <intense conflict resolution>
 $ git commit
 $ git push origin HEAD:improvement/GH-637-blocking-lint-and-more-lint-targets
```

Commit:
ci: remove python linting until we use python3 with salt

Issue: GH-637

Commit:
ci: introduce minimal YAML linting

Issue: GH-637

Commit:
ci: fix YAML linting errors

Commit:
ci: introduce minimal shell linting with `bashate`

Issue: GH-637

Commit:
ci: fix shell linting errors

Ignore E006 - line too long on bootstrap.spec.in,
needed by heredocs in this script.

Issue: GH-637

Commit:
ci: make lint errors blocking in CI

Issue: GH-637

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @cmonfort

* @alxf

* @jbertran

* @ballot-scality

* @carlito-scality

* @nicolast

* @tcarmet

* @zempashi

* @sayf-eddine-scality

* @gdemonet

* @monpote

* @alexandre-allard-scality

* @lucieleonard

* @slaperche-scality

* @thomasdanan

* @ramanduh

* @nootal

* @teddyandrieux

gdemonet commented on tox.ini:
```diff
@@ -2,21 +2,20 @@
+deps =
+    bashate==0.6.0
+commands =
+    bash -c "bashate **/*.sh"
```
Would be nice to add `**/*.sh.in` as well, so we can lint e.g. `scripts/bootstrap.sh.in`

jbertran replied:
Done. We ignore long line warnings for `.sh.in`, since they're needed by some of the heredocs included.

bert-e commented:
# Conflict




There is a conflict between your branch `improvement/GH-637-blocking-lint-and-more-lint-targets` and the
destination branch `development/2.0`.

Please resolve the conflict on **the feature branch** (`improvement/GH-637-blocking-lint-and-more-lint-targets`).

```bash
 $ git fetch
 $ git checkout origin/improvement/GH-637-blocking-lint-and-more-lint-targets
 $ git merge origin/development/2.0
 $ # <intense conflict resolution>
 $ git commit
 $ git push origin HEAD:improvement/GH-637-blocking-lint-and-more-lint-targets
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @cmonfort

* @alxf

* @jbertran

* @ballot-scality

* @carlito-scality

* @nicolast

* @tcarmet

* @zempashi

* @sayf-eddine-scality

* @gdemonet

* @monpote

* @alexandre-allard-scality

* @lucieleonard

* @slaperche-scality

* @thomasdanan

* @ramanduh

* @nootal

* @teddyandrieux

jbertran commented:
@bert-e approve

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @cmonfort

* @alxf

* @jbertran

* @ballot-scality

* @carlito-scality

* @nicolast

* @tcarmet

* @zempashi

* @sayf-eddine-scality

* @gdemonet

* @monpote

* @alexandre-allard-scality

* @lucieleonard

* @slaperche-scality

* @thomasdanan

* @ramanduh

* @nootal

* @teddyandrieux










*The following options are set:* **approve**

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.0`



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.0`



The following branches have **NOT** changed:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`



Please check the status of the associated issue GH-637.

Goodbye jbertran.
